### PR TITLE
NXBT-2542: update to node 10

### PR DESCRIPTION
**Pull request by nelsonsilva**

### CODE REVIEW

File: roles/slave_tools/tasks/nodejs.yml [hunk 1]
```diff
@@ -132,19 +132,52 @@
...
+  environment:
+    PATH: /opt/build/tools/node-v10.12.0-linux-x64/bin:{{ansible_env.PATH}}
+- name: Install Node 10.12 dependencies
+  npm: name={{item.name}} version={{item.version}} global=yes
+  with_items:
+  - name: yo
+    version: 2.0.5
+  - name: grunt-cli
+    version: 1.3.1
+  - name: gulp-cl1
```
Comment (woodyrew) [inline roles/slave_tools/tasks/nodejs.yml]: should this be `gulp-cli`?

File: roles/slave_tools/tasks/nodejs.yml [hunk 2]
```diff
@@ -132,19 +132,52 @@
...
+    version: 2.0.1
+  - name: bower
+    version: 1.8.4
+  - name: unicode
+    version: 11.0.1
```
Comment (woodyrew) [inline roles/slave_tools/tasks/nodejs.yml]: Do we need unicode as a globally installed package? It doesn't have an CLI API.

File: roles/slave_tools/tasks/nodejs.yml (hunk 1 above)
Comment (nelsonsilva) [reply roles/slave_tools/tasks/nodejs.yml]: afaik gulp should be installed locally, it's installed globally just to have the cli available system wide for which gulp-cli is enough. This is also following install instructions found at https://gulpjs.com/
I might review what we did for previous node versions but i'll do it in a separate PR (if ever :P )

File: roles/slave_tools/tasks/nodejs.yml (hunk 2 above)
Comment (nelsonsilva) [reply roles/slave_tools/tasks/nodejs.yml]: This is one of those "if it ain't broken don't fix it" situations.. tbh I just kept same set of global packages

File: roles/slave_tools/tasks/nodejs.yml (hunk 1 above)
Comment (woodyrew) [reply roles/slave_tools/tasks/nodejs.yml]: I was referring to the potential typo: `gulp-cl1` - gulp-cl one.
In my opinion, they should all be installed locally and run via scripts, npx or the local `npm bin` directory but that's another discussion.

**Comment by nelsonsilva** [comment]
Ran `mvn install -DskipTests`
(https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-nsilva/73)

**Pull request merged**